ddtrace/tracer/spancontext.go

```diff
@@ -330,7 +330,17 @@ func (t *trace) finishedOne(s *span) {
 			atomic.AddUint64(&tr.droppedP0Spans, uint64(len(t.spans)))
 			atomic.AddUint64(&tr.droppedP0Traces, 1)
 		}
-		return
+		// if trace sampling decision is drop, we still want to send single spans
+		var singleSpans []*span
+		for _, span := range t.spans {
+			if tr.rulesSampling.singleSpanRulesSampler.apply(span) {
```

Comment:
```suggestion
			if tr.rulesSampling.sampleSpan(span) {
```
The unexported field of `rulesSampling` should not be known to the caller. It will make for a friendlier API. That was the whole purpose.